feat(icons): renaming diamond status icon glyph

conradennis opened the pull request:
Old version of the icon was named `undefined.svg` which was conflicting with another icon in the repo.

Renaming to `diamond-fill.svg` to match the convention of other status glyph names and resolve the naming conflict.

Updated categories.yml file with new icon name, alphabetized out of place icon names.
Approved by Carbon and Brand team

#### Changelog

**New name**

- svg/diamond-fill.svg

**Old name**

- svg/undefined.svg

conradennis committed:
feat(icons): renaming diamond status icon glyph

Old version of the icon was named "undefined" which was conflicting with another icon in the repo.

Renaming to "diamond-fill" to match the convention of other status glyph names and resolve the naming conflict.

Updated categories.yml file with new icon name.
Approved by Carbon and Brand team

alisonjoseph committed:
Merge branch 'main' into diamond-status-icon-update

alisonjoseph reviewed (changes requested):
Hey Conrad, I think you'll also need to add the Diamond fill icon to the `icons.yml `file, and remove the `- glyph` for the undefined icon in the same file since that no longer exists. Let me know if that doesn't make sense.

conradennis committed:
added diamond-fill to icons.yml file

added diamond-fill to icons.yml file and removed glyph sizing from undefined.svg icon

conradennis committed:
Merge branch 'diamond-status-icon-update' of https://github.com/conradennis/carbon into diamond-status-icon-update

alisonjoseph committed:
chore: update snapshots

tw15egan committed:
Merge branch 'main' into diamond-status-icon-update

tay1orjones committed:
Merge branch 'main' into diamond-status-icon-update

kodiakhq[bot] committed:
Merge branch 'main' into diamond-status-icon-update
